marine/marine.py

```diff
@@ -16,7 +31,73 @@ class MarineResult(Structure):
     _fields_ = [("output", POINTER(c_char_p)), ("len", c_uint), ("result", c_int)]
 
 
+class MarinePacketFieldValues(ctypes.Union):
+    pass
+
+
+class MarinePacketFieldValue(Structure):
+    pass
+
+
+MarinePacketFieldValues._fields_ = [
+    ("float_value", c_double),
+    ("int_value", c_int),
+    ("uint_value", c_uint),
+    ("bool_value", c_ubyte),
+    ("str_value", c_char_p),
+    ("bytes_value", c_char_p),
+    ("list_value", POINTER(MarinePacketFieldValue)),
+]
+
+MarinePacketFieldValue._anonymous_ = ["values"]
+MarinePacketFieldValue._fields_ = [
+    ("values", MarinePacketFieldValues),
+    ("len", c_uint),
+    ("type", c_uint),
+]
+
+
+class MarinePacketFieldValueType:
```

Comment:
Can you use the enum that is defined in `marine.h` for this? Similar to how we use `MARINE_ALREADY_INITIALIZED_ERROR_CODE` for example.

Reply:
Sadly, this doesn't seem to be very trivial :(
The value you referenced, "MARINE_ALREADY_INITIALIZED_ERROR_CODE` is an int const, while `marine_value_type` is an enum, and I would prefer it to remain that way (separate type, automatic index generation etc).
I didn't find a simple way to expose all values of this enum from the shared library (I even found some resources that outright suggest duplicating the enum), So doing this would most likely be the lesser evil in this situation.